extras/api/views.py

```diff
@@ -181,3 +191,195 @@ class WebhookViewSet(PeeringManagerModelViewSet):
     queryset = Webhook.objects.all()
     serializer_class = WebhookSerializer
     filterset_class = WebhookFilterSet
+
+
+class BGPq4ViewSet(ViewSet):
+    """
+    ViewSet for BGPq4 prefix queries using existing Redis cache and BGPq infrastructure
+    """
+
+    permission_classes = [IsAuthenticated]
```

Comment:
If login is not required the endpoint should be usable so `IsAuthenticatedOrLoginNotRequired` would be more suited.